nvflare/app_common/widgets/streaming.py

```diff
@@ -170,13 +268,14 @@ def handle_event(self, event_type: str, fl_ctx: FLContext):
             record_origin = fl_ctx.get_identity_name()
 
             # if fed event use peer name to save
-            if data.get_header(FedEventHeader.ORIGIN) is not None:
+            if fl_ctx.get_prop(FLContextKey.EVENT_SCOPE) == EventScope.FEDERATION:
                 peer_name = data.get_peer_prop(ReservedKey.IDENTITY_NAME, None)
                 record_origin = peer_name
```

Comment:
These lines are OK.  It might be more clear if we say
if fl_ctx.get_prop(....) == ... :
  record_origin = data.get_peer_prop(....)
else:
  record_origin = fl_ctx.get_identity_name()

Reply:
No need to submit another commit.  I will mark this as resolved.